### PR TITLE
Add cloudauthz?

### PR DESCRIPTION
**Pull request by dannon**

### CODE REVIEW

**Comment by nsoranzo** [comment]
@natefoo I think you need to rebase on master because of my changes.

**Comment by dannon** [comment]
Well, I took a quick look to see if I could do it, but I'm not sure how to handle the matrix collision.  Do we really need separate builds for 3.4, 3.5, 3.6, 3.7?

**Comment by nsoranzo** [comment]
> Well, I took a quick look to see if I could do it, but I'm not sure how to handle the matrix collision. Do we really need separate builds for 3.4, 3.5, 3.6, 3.7?

I think we do for packages that are not pure Python, which @natefoo disabled here only for testing.

**Comment by natefoo** [comment]
@nsoranzo Thanks, yeah, I was going to look at that once I got this working.

**Comment by natefoo** [comment]
> I think we do for packages that are not pure Python, which @natefoo disabled here only for testing.

Exactly this.

**Comment by dannon** [comment]
Ok, cool.  I'll do other things, then, thanks!  :)

File: .ci/wheels-install.sh [hunk 1]
```diff
@@ -28,18 +28,23 @@ case $TRAVIS_OS_NAME in
...
         curl -LO https://www.python.org/ftp/python/${PY}.${pypt}/python-${PY}.${pypt}-macosx10.6.pkg
         sudo installer -pkg python-${PY}.${pypt}-macosx10.6.pkg -target /
-        curl -O "$virtualenv"
-        tar zxvf virtualenv-${virtualenv_vers}.tar.gz
-        /Library/Frameworks/Python.framework/Versions/${PY}/bin/python${PY%%.*} ./virtualenv-${virtualenv_vers}/virtualenv.py $HOME/venv
-        . $HOME/venv/bin/activate
+        #curl -O "$virtualenv"
+        #tar zxvf virtualenv-${virtualenv_vers}.tar.gz
+        #/Library/Frameworks/Python.framework/Versions/${PY}/bin/python${PY%%.*} ./virtualenv-${virtualenv_vers}/virtualenv.py $HOME/venv
+        # we need Starforge in the buildenv since the osx build uses the local execucution context, but installing it a
```
Comment (nsoranzo) [inline .ci/wheels-install.sh]: s/execucution/execution/

File: .travis.yml [hunk 2]
```diff
@@ -1,36 +1,115 @@
...
+        wheel_type=$($STARFORGE_CMD wheel_type --wheels-config="$meta" "$wheel")
+        if [ "$wheel_type" == "$WHEEL_BUILDER_TYPE" ]; then
+            BUILD_WHEEL_METAS+=("$meta")
+        else
+            echo "Builder for '$WHEEL_BUILDER_TYPE' skipping wheel '$wheel' of type '$wheel_type'"
+        fi
+    done
+  - set | grep '^BUILD_WHEEL_METAS='
+  - "[ ${#BUILD_WHEEL_METAS[@]} -gt 0 ] || { echo \"No wheel changes for builder '$WHEEL_BUILDER_TYPE', terminating\"; travis_terminate 0; }"
+  - "[ $WHEEL_BUILDER_TYPE != 'c-extension' ] || ./.ci/wheels-install.sh"
```
Comment (nsoranzo) [inline .travis.yml]: You can move `./.ci/wheels-install.sh` inside the `if` branch at line 86.

File: .ci/wheels-deploy.sh [hunk 3]
```diff
@@ -2,6 +2,11 @@
 set -e
```
Comment (nsoranzo) [inline .ci/wheels-deploy.sh]: I think this file is unused now, remove?

**Comment by natefoo** [comment]
@nsoranzo thanks, still WIP (but getting close now!)

Interestingly since rebasing, `git diff ... "$TRAVIS_COMMIT_RANGE"` no longer indicates a change to cloudauthz `meta.yml` so now nothing runs. Once this is done I'm going to commit all of the Travis changes (everything not having to do with cloudauthz) to master and rebase this branch on top of it so it only includes the cloudauthz addition. Hopefully that'll fix it.

**Comment by nsoranzo** [comment]
> Interestingly since rebasing, git diff ... "$TRAVIS_COMMIT_RANGE" no longer indicates a change to cloudauthz meta.yml so now nothing runs.

@natefoo You need to look at the Travis build for `pr`, not `push`, e.g. in https://travis-ci.org/galaxyproject/starforge-recipes/builds/419859747 jobs 37.12 and 37.13 compiled cloudauthz.

**Comment by natefoo** [comment]
Whoops, good catch (as always!) :cry:

File: wheels/cloudauthz/meta.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+
+type: wheel
+name: cloudauthz
+purepy: true
```
Comment (natefoo) [inline wheels/cloudauthz/meta.yml]: This is no longer needed unless Starforge fails to autodetect the wheel type.

File: wheels/cloudauthz/meta.yml [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+version: 0.1.0
+src:
+    - https://files.pythonhosted.org/packages/ac/44/20d0e6865db1fafc6d3f0fd2ee767d42e287914be5e4d20649a22d1494b6/cloudauthz-0.1.0.tar.gz
```
Comment (natefoo) [inline wheels/cloudauthz/meta.yml]: Also not needed as long as `pip download <package>` is all you need to fetch the package source (so anything in PyPI).

File: wheels/cloudauthz/meta.yml (hunk 4 above)
Comment (dannon) [reply wheels/cloudauthz/meta.yml]: Cool

File: wheels/cloudauthz/meta.yml (hunk 5 above)
Comment (dannon) [reply wheels/cloudauthz/meta.yml]: @natefoo Even better!

**Comment by natefoo** [comment]
@dannon Once everything works I'll update the Starforge docs to explain the `meta.yml` attributes.

**Comment by natefoo** [comment]
@nsoranzo you were right about the `travis_terminate` hang on `osx` only happening in the `before_script` stage. This is going to add needless complexity to the later stages if it really can't be terminated here. :frowning_face:

**Comment by nsoranzo** [comment]
Superseded by #10.

**Pull request closed without merging**